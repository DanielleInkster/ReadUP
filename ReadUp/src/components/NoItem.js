import React from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';

const NoItem = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.heading}>Nothing to see here - yet! </Text>
      <Text style={styles.text}>Press the button to start adding articles. </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '95%',
    marginTop: '5%',
    marginBottom: '5%',
    alignSelf: 'center',
    backgroundColor: 'rgba(187,225,250,0.88)',
    borderRadius: 15,
  },

  heading: {
    fontSize: 30,
    fontWeight: 'bold',
    fontStyle: 'italic',
    textAlign: 'center',
    marginTop: '70%',
    color: '#0f4c75',
  },
  text: {
    fontSize: 20,
    fontStyle: 'italic',
    textAlign: 'center',
    marginTop: '5%',
    color: '#0f4c75',
  },
});

export default NoItem;
