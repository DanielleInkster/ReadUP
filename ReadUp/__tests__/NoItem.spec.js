import React from 'react';
import {render} from 'react-native-testing-library';
import NoItem from '../src/components/NoItem';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<NoItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('NoItem', () => {
  it('displays startup info', () => {
    const {getByText} = render(<NoItem />);
    expect(getByText('Nothing to see here - yet! ')).not.toBeNull();
    expect(
      getByText("Press the 'Edit' button to start adding articles."),
    ).not.toBeNull();
  });
});
