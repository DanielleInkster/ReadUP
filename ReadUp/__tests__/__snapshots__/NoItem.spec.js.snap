// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "rgba(187,225,250,0.88)",
      "borderRadius": 15,
      "flex": 1,
      "marginBottom": "20%",
      "marginTop": "5%",
      "width": "95%",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "alignSelf": "flex-end",
          "color": "#0f4c75",
          "fontSize": 75,
          "paddingRight": "5%",
          "paddingTop": "5%",
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï…¶
  </Text>
  <Text
    style={
      Object {
        "color": "#0f4c75",
        "fontFamily": "sans-serif-medium",
        "fontSize": 30,
        "fontStyle": "italic",
        "fontWeight": "bold",
        "marginTop": "30%",
        "textAlign": "center",
      }
    }
  >
    Nothing to see here - yet! 
  </Text>
  <Text
    style={
      Object {
        "color": "#0f4c75",
        "fontFamily": "sans-serif-medium",
        "fontSize": 25,
        "fontStyle": "italic",
        "marginTop": "5%",
        "textAlign": "center",
      }
    }
  >
    Press the 'Edit' button to start adding articles.
  </Text>
</View>
`;
